core/src/main/resources/hudson/PluginManager/_table.js

```diff
@@ -10,46 +10,214 @@ function checkPluginsWithoutWarnings() {
-                var part = filterParts[j];
-                if (content.indexOf(part) < 0) {
-                    makeVisible = false;
-                    break;
-                }
-            }
-            if (makeVisible) {
-                items[i].removeClassName("hidden");
-                anyVisible = true;
+        clearTimeout(debounce);
+        // debounce reduces number of server side calls while typing
+        debounce = setTimeout(function() {
+            var pluginsTable = document.getElementById('plugins');
+
+            // only on available tab
+            if (pluginsTable.dataset.page === 'available') {
+                view.availablePlugins(e.value.toLowerCase().trim(), 20, function (plugins) {
+                    var availablePlugins = JSON.parse(plugins.responseObject());
+
+                    var tbody = pluginsTable.querySelector('tbody');
+
+                    function clearOldResults() {
+                        if (pluginsTable.dataset.hasadmin !== 'true') {
+                            tbody.innerHTML = '';
+                        } else {
+                            var rows = tbody.querySelectorAll('tr');
+                            if (rows) {
+                                rows.forEach(function (row) {
+                                    var input = row.firstChild.firstChild;
+                                    if (input.checked !== true) {
+                                        row.remove();
+                                    }
+                                })
+                            }
+                        }
+                    }
+
+                    clearOldResults();
+
+                    availablePlugins.forEach(function (plugin) {
+                        var htmlTableRowElement = document.createElement('tr');
+                        htmlTableRowElement.classList.add('plugin');
+
+                        function createCheckBox() {
+                            var htmlTableDataCellElement = document.createElement('td');
+                            htmlTableDataCellElement.classList.add('pane')
+                            var htmlInputElement = document.createElement('input');
+                            htmlInputElement.type = 'checkbox';
+                            htmlInputElement.name = 'plugin.' + plugin.name + '.' + plugin.sourceId;
+                            htmlTableDataCellElement.append(htmlInputElement);
+                            htmlTableRowElement.append(htmlTableDataCellElement);
+                        }
+
+                        function createNameEntry() {
+                            var htmlTableDataCellElement = document.createElement('td');
+                            htmlTableDataCellElement.classList.add('pane')
+                            var htmlDivElement = document.createElement('div');
+                            var htmlAnchorElement = document.createElement('a');
+                            htmlAnchorElement.href = plugin.wiki
```

Comment:
so technically this is controlled by the pom right? so technically they could have a wiki link of `javascript:alert("foo")`? do we need to sanitize this at all? make sure its a parsed url in the backend?

Reply:
Isn’t that done in update center? This is just a port of the code from the jelly files

Reply:
I have no idea. I'm not super familiar with core, its a common javascript issue so I just wanted to point it out just in case.

Reply:
@daniel-beck ?

Reply:
Maybe `setAttribute` is better?

Reply:
Not really..any time you use user values it has the potential to inject/XSS. I think the attack vector is pretty small. The best solution is to use new Uri in java and confirm the scheme is http or HTTPS. Then after that no concern.

Reply:
it's hardcoded in update center to plugin site: https://github.com/jenkins-infra/update-center2/blob/4c19fc6ca406abb5dc37adf7aa311d58b98d2963/src/main/java/io/jenkins/update_center/PluginUpdateCenterEntry.java#L92